src/PKSim.Core/Model/CompoundPK.cs

```diff
@@ -1,37 +1,89 @@
-﻿using OSPSuite.Utility.Collections;
+using System.Linq;
+using OSPSuite.Core.Domain;
+using OSPSuite.Utility.Collections;
 
 namespace PKSim.Core.Model
 {
    /// <summary>
-   /// Reperesents a cache containing specific PK-Values calculated for a given compound
+   /// Represents a cache containing specific PK-Values calculated for a given compound
    /// </summary>
    public class CompoundPK
    {
+      private readonly Cache<string, QuantityPKParameter> _quantityPKParameters = new Cache<string, QuantityPKParameter>(getKey:x => x.Name);
+
+      public CompoundPK()
+      {
+         bioAvailabilities = new Cache<int, double?>();
+         ddiAucInf = new Cache<int, double?>();
+         ddiCMax = new Cache<int, double?>();
+      }
       public string CompoundName { get; set; }
-      
+
       /// <summary>
-      /// AUC_inf of plasma curve following a IV infustion of 15 min
+      /// AUC_inf of plasma curve following a IV infusion of 15 min
       /// </summary>
-      public double? AucIV { get; set; }
+      private Cache<int, double?> bioAvailabilities { get; set; }
 
       /// <summary>
       /// AUC_inf of plasma curve following an application where only <see cref="CompoundName"/> is applied
       /// </summary>
-      public double? AucDDI { get; set; }
+      private Cache<int, double?> ddiAucInf { get; set; }
+      private Cache<int, double?> ddiCMax { get; set; }
 
-      /// <summary>
-      /// Cmax of plasma curve following an application where only <see cref="CompoundName"/> is applied
-      /// </summary>
-      public double? CmaxDDI { get; set; }
+      private double? valueStoreFor(ICache<int, double?> allValues, int individualId)
+      {
+         if (!allValues.Contains(individualId))
+            allValues[individualId] = null;
+
+         return allValues[individualId];
+      }
+
+      public ICache<int, double?> AllBioAvailabilities => bioAvailabilities;
```

Comment:
you can probably crate and init in one go here .This is probably public for serializatiohn